api/src/main/java/net/kyori/adventure/util/internal/AdventureProperties.java

```diff
@@ -0,0 +1,118 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package net.kyori.adventure.util.internal;
+
+import java.util.function.Function;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Adventure properties.
+ *
+ * @since 4.10.0
+ */
+@ApiStatus.Internal
+public final class AdventureProperties {
+  /**
+   * Option for specifying the default translation locale.
+   *
+   * @since 4.10.0
+   */
+  public static final Option<String> DEFAULT_TRANSLATION_LOCALE = option("defaultTranslationLocale", Function.identity());
+  /**
+   * Option for specifying whether service load failures are fatal.
+   *
+   * @since 4.10.0
+   */
+  public static final Option<Boolean> SERVICE_LOAD_FAILURES_ARE_FATAL = option("serviceLoadFailuresAreFatal", Boolean::parseBoolean);
+  /**
+   * Option for specifying whether to warn when legacy formatting is detected.
+   *
+   * @since 4.10.0
+   */
+  public static final Option<Boolean> TEXT_WARN_WHEN_LEGACY_FORMATTING_DETECTED = option("text.warnWhenLegacyFormattingDetected", Boolean::parseBoolean);
+
+  private AdventureProperties() {
+  }
+
+  /**
+   * Creates a new option.
+   *
+   * @param name the option name
+   * @param parser the value parser
+   * @param <T> the value type
+   * @return the option
+   * @since 4.10.0
+   */
+  public static <T> @NotNull Option<T> option(final @NotNull String name, final @NotNull Function<String, T> parser) {
+    return new AdventurePropertiesImpl.OptionImpl<>(name, parser);
+  }
+
+  /**
+   * Gets a boolean value.
+   *
+   * @param option the option
+   * @param defaultValue the default value
+   * @return the boolean value
+   * @since 4.10.0
+   */
+  public static boolean booleanValueOf(final @NotNull Option<Boolean> option, final boolean defaultValue) {
+    return Boolean.TRUE.equals(valueOf(option, defaultValue));
+  }
```

Comment:
What about just having a `get()` or `value()` like kezz talked about in another comment in `Option` instead?

That way 
```java
AdventureProperties.booleanValueOf(AdventureProperties.TEXT_WARN_WHEN_LEGACY_FORMATTING_DETECTED, false);
```
could become
```java
AdventureProperties.TEXT_WARN_WHEN_LEGACY_FORMATTING_DETECTED.value();
```

The default value could be decided when each `Option` is parsed.
```java
public static final Option<Boolean> TEXT_WARN_WHEN_LEGACY_FORMATTING_DETECTED = option("text.warnWhenLegacyFormattingDetected", Boolean::parseBoolean, false);
```